set.mm

```diff
@@ -127214,6 +127214,34 @@ Rational numbers (as a subset of complex numbers)
     LVOVHVRQVKVHVOVRVHVDVEVRVFBUCRUDUEVPVQADVKVDVEVQAUFZVDVKBSFZVEVSBUKVKASFVTV
     EVSAULABUGUHUIRUJUMUNUOUPUQURUSUTVFCDMVA $.
 
+  ${
+    $d A x y z $.
+    $( A positive rational is the quotient of two positive integers.
+       (Contributed by AV, 29-Dec-2022.) $)
+    elpq $p |- ( ( A e. QQ /\ 0 < A )
+                 -> E. x e. NN E. y e. NN A = ( x / y ) ) $=
```

Comment:
You can inline it in the proof of elpqb since it is subsumed by it.
You probably know that but just in case: do `MM> PROVE elpqb` and then `MM-PA> EXPAND elpq`, save the proof , write to source, and then remove elpq.

Reply:
I want to keep ~elpq as a separate theorem, because it is the more difficult part of the bijection (takes about 2/3 of the inlined proof). I could have made it a lemma (for ~elpqb ), but I think that one of the directions of a bijection deserves to be a theorem. 

Once there was a discussion that long proofs are difficult to understand (and to be minimized automatically), and somebody suggested to provide additional means to structure a proof. David and I argued that this is not needed, because interesting parts can (and should) be singled out as separate theorems, or as lemmas at least.

Reply:
I think the combined proof would not be very long either, and the last step being biimpi, it is easy to see the two parts on the webpage. To me it is superfluous to have both a biconditional statement and one of its implications (with some exceptions, for instance for the simplest propcalc theorems, or when one direction requires fewer axioms), and there are many library theorems that accept biconditionals when only one implication is needed (mpbi, sylbi...).

I do not want to appear dictatorial, so if you do not agree, let's hear from a third person (and since @jkingdon was around...).

Reply:
For me it is mostly a question of how long the proofs are (with a certain bias towards separating out theorems which can be readily stated separately, such as the two directions of a biconditional).

In this case, the one direction is 35 steps and the other 17 so I'd separate them. I might call them `elpq` (for the biconditional) and `elpqlem` (or something else starting with `elpqlem`) for the direction which is split out, but that is a minor non-blocking point and a few comments up @avekens expresses a preference not to do this, so I'd go with the author's preference on a thing like that.

Reply:
Since @jkingdon is at least not against a separate theorem elpq, I think we can keep it (and merge this PR).

Another (minor) reason to keep it is the comment "A positive rational is the quotient of two positive integers." is easier to understand (for non-formalists) than "A class is a positive rational...".